## Sleep & Lifestyle Dataset Kaggle

- I used this video as a guide: https://www.youtube.com/watch?v=xi0vhXFPegw
- Here's the dataset I imported: https://www.kaggle.com/datasets/ayeshaimran1619/sleep-and-lifestyle-study

In this exercise I'm going to:
- import and read the data
- understand the data
- do data preparation
- feature understanding
- feature relationships

In [28]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [29]:
sl_data = pd.read_csv("C:/Users/Bobo/Downloads/Sleep_health_and_lifestyle_dataset.csv")

In [30]:
sl_data.head()

,Person ID,Gender,Age,Occupation,Sleep Duration,Quality of Sleep,Physical Activity Level,Stress Level,BMI Category,Blood Pressure,Heart Rate,Daily Steps,Sleep Disorder
0,1,Male,27,Software Engineer,6.1,6,42,6,Overweight,126/83,77,4200,NaN
1,2,Male,28,Doctor,6.2,6,60,8,Normal,125/80,75,10000,NaN
2,3,Male,28,Doctor,6.2,6,60,8,Normal,125/80,75,10000,NaN
3,4,Male,28,Sales Representative,5.9,4,30,8,Obese,140/90,85,3000,Sleep Apnea
4,5,Male,28,Sales Representative,5.9,4,30,8,Obese,140/90,85,3000,Sleep Apnea


In [31]:
sl_data.shape

(374, 13)

In [32]:
sl_data.columns

Index(['Person ID', 'Gender', 'Age', 'Occupation', 'Sleep Duration',
       'Quality of Sleep', 'Physical Activity Level', 'Stress Level',
       'BMI Category', 'Blood Pressure', 'Heart Rate', 'Daily Steps',
       'Sleep Disorder'],
      dtype='object')

#### Example of Modifying our dataset (dropping columns)

Let's say we decided we don't want to use all of these columns now and we're only interested in a few of them. We'll create a new (sub-dataset), which will only include the necessary ones

In [33]:
sl_data_short = sl_data[['Person ID', 'Gender', 'Age', 'Occupation', 'Sleep Duration',
       #'Quality of Sleep', 
         'Physical Activity Level', 
         'Stress Level',
       'BMI Category' #'Blood Pressure', 'Heart Rate', 'Daily Steps',
       #'Sleep Disorder'
        ]]

We see that we still have 374 rows but this time we have only 8 columns (the ones we want to use further in our analysis)

In [34]:
sl_data_short.dtypes

Person ID                    int64
Gender                      object
Age                          int64
Occupation                  object
Sleep Duration             float64
Physical Activity Level      int64
Stress Level                 int64
BMI Category                object
dtype: object

Everything seems pretty normal here in terms of data formats so we're going to proceed with another task.

### Renaming columns
Let's say we're not happy with the labels of some of our columns

In [35]:
sl_data_short.head()

,Person ID,Gender,Age,Occupation,Sleep Duration,Physical Activity Level,Stress Level,BMI Category
0,1,Male,27,Software Engineer,6.1,42,6,Overweight
1,2,Male,28,Doctor,6.2,60,8,Normal
2,3,Male,28,Doctor,6.2,60,8,Normal
3,4,Male,28,Sales Representative,5.9,30,8,Obese
4,5,Male,28,Sales Representative,5.9,30,8,Obese


In [36]:
# Let's say we don't like the "Sleep Duration" column name:
sl_data_short.columns # we use this column command once again just to see what the names of our columns looked like exactly

Index(['Person ID', 'Gender', 'Age', 'Occupation', 'Sleep Duration',
       'Physical Activity Level', 'Stress Level', 'BMI Category'],
      dtype='object')

In [37]:
sl_data_short = sl_data_short.rename(columns={'Sleep Duration': 'Sleep_Dur',
                     'Physical Activity Level': 'Activity_Level'})

In [38]:
sl_data_short

,Person ID,Gender,Age,Occupation,Sleep_Dur,Activity_Level,Stress Level,BMI Category
0,1,Male,27,Software Engineer,6.1,42,6,Overweight
1,2,Male,28,Doctor,6.2,60,8,Normal
2,3,Male,28,Doctor,6.2,60,8,Normal
3,4,Male,28,Sales Representative,5.9,30,8,Obese
4,5,Male,28,Sales Representative,5.9,30,8,Obese
...,...,...,...,...,...,...,...,...
369,370,Female,59,Nurse,8.1,75,3,Overweight
370,371,Female,59,Nurse,8.0,75,3,Overweight
371,372,Female,59,Nurse,8.1,75,3,Overweight
372,373,Female,59,Nurse,8.1,75,3,Overweight
